detector.go

```diff
@@ -58,30 +58,30 @@ type Detector struct {
 	DirStore       DirStore
 	Executor       buildpack.DetectExecutor
 	HasExtensions  bool
-	Logger         log.LoggerWriterWithLevel
-	MultiLogger    *log.MultiLogger
+	Logger         log.LoggerHandlerWithLevel
+	MemHandler     *memory.Handler
 	Order          buildpack.Order
 	PlatformDir    string
 	Resolver       DetectResolver
 	Runs           *sync.Map
 }
 
-func (f *DetectorFactory) NewDetector(appDir, buildConfigDir, orderPath, platformDir string, logger log.LoggerWriterWithLevel) (*Detector, error) {
+func (f *DetectorFactory) NewDetector(appDir, buildConfigDir, orderPath, platformDir string, logger log.LoggerHandlerWithLevel) (*Detector, error) {
+	memHandler := memory.New()
 	detector := &Detector{
 		AppDir:         appDir,
 		BuildConfigDir: buildConfigDir,
 		DirStore:       f.dirStore,
 		Executor:       &buildpack.DefaultDetectExecutor{},
 		Logger:         logger,
 		PlatformDir:    platformDir,
+		Resolver:       NewDefaultDetectResolver(&apexlog.Logger{Handler: memHandler}),
 		Runs:           &sync.Map{},
+		MemHandler:     memHandler,
```

Comment:
I think we could straight up hide `MemHandler` and let it get built and stored in `Detector`. It isn't something a caller needs to send in. More of an implementation detail IMO.

Reply:
You're right... updated in 04799076